test/test_cpp_standards.py

```diff
@@ -35,15 +114,32 @@ def test_cpp14():
     assert set(actual_wheels) == set(expected_wheels)
 
 
-def test_cpp17():
+cpp17_project = cpp_template_project.copy()
+
+if utils.platform == 'windows':
+    cpp17_project.template_context['extra_compile_args'] = ['/std:c++17', '/wd5033']
+else:
+    cpp17_project.template_context['extra_compile_args'] = ['-std=c++17', '-Wno-register']
+
+cpp17_project.template_context['spam_cpp_top_level_add'] = r'''
+#include <utility>
+auto a = std::pair(5.0, false);
+'''
+
+
+def test_cpp17(tmp_path):
     # This test checks that the C++17 standard is supported
+    project_dir = tmp_path / 'project'
+
+    cpp17_project.generate(project_dir)
 
     # Python and PyPy 2.7 use the `register` keyword which is forbidden in the C++17 standard
     # The manylinux1 docker image does not have a compiler which supports C++11
+    # Python 3.5 and PyPy 3.6 are compiled with MSVC 10, which does not support C++17
```

Comment:
This line got added again during the merge. It's not true, anyway. It's MSVC 14 (VS 2015)